provers/evm-prover/src/main.rs

```diff
@@ -0,0 +1,191 @@
+use std::env;
+use std::fs;
+use tonic::{transport::Server, Request, Response, Status};
+
+// Import the generated proto rust code
+pub mod prover {
+    tonic::include_proto!("celestia.prover.v1");
+}
+
+use prover::prover_server::{Prover, ProverServer};
+use prover::{
+    ProveMembershipRequest, ProveMembershipResponse, ProveStateTransitionRequest,
+    ProveStateTransitionResponse,
+};
+
+use celestia_rpc::{BlobClient, Client, HeaderClient};
+use celestia_types::{nmt::Namespace, Blob};
+use rsp_client_executor::io::ClientExecutorInput;
+use sp1_sdk::{include_elf, ProverClient, SP1Stdin};
+
+use ethers::{
+    providers::{Http, Middleware, Provider},
+    types::BlockNumber,
+};
+
+/// The ELF file for the Succinct RISC-V zkVM.
+const BLEVM_ELF: &[u8] = include_elf!("blevm");
+
+pub struct ProverService {
+    celestia_client: Client,
+    evm_client: Provider<Http>,
+    sp1_prover: ProverClient,
+}
+
+impl ProverService {
+    async fn new() -> Result<Self, Box<dyn std::error::Error>> {
+        let token = env::var("CELESTIA_NODE_AUTH_TOKEN").expect("Token not provided");
```

Comment:
nit: this ENV variable isn't described in the README yet

Reply:
Updated.